PeripheryFunctions/TS_GetIDs.m

```diff
@@ -67,29 +73,27 @@
 case 'ops'
     theDataTable = Operations;
 otherwise
-    error('Specify ''ts'' or ''ops''');
+    error('Specify ''ts'', ''ops'', or ''opsName''');
 end
 
-% The cell of comma-delimited keyword strings:
-theKeywordCell = theDataTable.Keywords;
-
-% Split into sub-cells using comma delimiter:
-Keywords = SUB_cell2cellcell(theKeywordCell);
-
-% Find objects with a keyword that matchees that given:
-matches = cellfun(@(x)any(ismember(theKeyword,x)),Keywords);
+% OC: below has been changed so that we return a set in the same order as
+%       above, and nan's otherwise
+matches = nan(length(theFields),1);
+for i = 1:length(theFields)
+    cmatch = find(strcmp(theDataTable.(nameOrKeywords),theFields{i}));
+    if ~isempty(cmatch)
+        matches(i) = cmatch;
+    end
+end
```

Comment:
This will not work for keywords that have duplicate text because it doesn't respect the comma formatting in the keywords, e.g., it just matches raw text of say 'condition1' will also match with 'condition10'…
Perhaps we can keep the original code for keywords, but add functionality for matching to name?

Reply:
Hmm, this is true. Perhaps when the `nameOrKeywords` parameter is set to `keyword`, we use the original method, and when set to `name`, it uses the new approach?

Probably better.. should we just create a new function? One that finds keywords (unsorted) and one that finds names (sorted)?

Reply:
Yes, the current version (commits I made yesterday) does this. Can you pull the latest and inspect? (Or take a look in the "Files changed" tab of this PR)